MAUI/DataGrid/export-to-excel.md

```diff
@@ -0,0 +1,548 @@
+    DataGridExcelExportingController excelExport = new DataGridExcelExportingController();
+    var excelEngine = excelExport.ExportToExcel(this.dataGrid);
+    var workbook = excelEngine.Excel.Workbooks[0];
+    MemoryStream stream = new MemoryStream();
+    workbook.SaveAs(stream);
+    workbook.Close();
+    excelEngine.Dispose();
+
+    string OutputFilename = "DefaultDataGrid.xlsx";
+    SaveService saveService = new();
+    saveService.SaveAndView(OutputFilename, "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", stream);
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to Excel format" src="Images\Export-to-Excel\maui-datagrid-datagrid-to-excel.png" width="689"/>
+
+N> SfDataGrid cannot export the DataGridTemplateColumn to PDF or Excel,  since we cannot get the loaded views and draw them with the particular range, values etc from DataGridTemplateColumn.
+
+## Exporting Options
+
+Additionally, you can export the data to Excel by providing the grid and `DataGridExcelExportingOption` as arguments to the `ExportToExcel` method, which offers several customization options.
+
+{% tabs %}
+{% highlight c# %}
+DataGridExcelExportingController excelExport = new DataGridExcelExportingController();
+DataGridExcelExportingOption exportOption = new DataGridExcelExportingOption();
+exportOption.CanExportColumnWidth = false;
+exportOption.DefaultColumnWidth = 150;
+var excelEngine = excelExport.ExportToExcel(this.dataGrid, exportOption);
+{% endhighlight %}
+{% endtabs %}
+
+The SfDataGrid provides several properties in `DataGridExcelExportingOption` class to customize the grid while exporting it to Excel.
+
+### Exporting formatted text and actual value
+
+By default, the actual value will only be exported to Excel. To export the display text, set the ExportMode property as Text.
+
+{% tabs%}
+{% highlight c# %}
+DataGridExcelExportingController excelExport = new DataGridExcelExportingController ();
+DataGridExcelExportingOption options = new DataGridExcelExportingOption ();
+options.ExportMode = ExportMode.Text;
+var excelEngine = excelExport.ExportToExcel(dataGrid, options);
+var workBook = excelEngine.Excel.Workbooks[0];
+{% endhighlight c# %}
+{% endtabs %}
+
+### Getting RowIndex, ColumnIndex and Columns for customization
+
+#### ExcelColumnIndex
+
+The `ExcelColumnIndex Property` gets or internally sets the column index being exported to the Excel. Each column is exported based on this index to identify the current exporting column index.
+
+#### ExcelRowIndex
+
+The `ExcelRowIndex Property` gets the row index being exported to the Excel. Each row is exported based on this index to identify the current exporting row index.
+
+#### Columns
+
+Using the property `System.Collections.IEnumerable Columns` you can get or set the `ExcludedColumns` columns collection which contains all the columns that are to be exported. The columns in the ExcludedColumns List will not be a member of the Columns collection.
```

Comment:
#### ExcelColumnIndex

The `ExcelColumnIndex` property retrieves or internally sets the column index that is being exported to Excel. Each column is exported based on this index to identify the current exporting column.

#### ExcelRowIndex

The `ExcelRowIndex` property retrieves the row index that is being exported to Excel. Each row is exported based on this index to identify the current exporting row index.

#### Columns

By using the `System.Collections.IEnumerable Columns` property, you can retrieve or set the `ExcludedColumns` collection of columns, which contains all the columns to be exported. The columns in the `ExcludedColumns` list will not be included in the `Columns` collection.

Reply:
updated